wandb/sdk/data_types/base_types/media.py

```diff
@@ -27,6 +27,40 @@
 def _wb_filename(
     key: Union[str, int], step: Union[str, int], id: Union[str, int], extension: str
 ) -> str:
+    """Generates a safe filename/path for storing media files, using the provided key, step, and id.
+
+    The filename is made safe by:
+    1. Removing any leading slashes to prevent writing to absolute paths
+    2. Replacing '.' and '..' with underscores to prevent directory traversal attacks
+
+    If the key contains slashes (e.g. 'images/cats/fluffy.jpg'), subdirectories will be created:
+        media/
+          images/
+            cats/
+              fluffy.jpg_step_id.ext
+
+    Args:
+        key: Name/path for the media file
+        step: Training step number
+        id: Unique identifier
+        extension: File extension (e.g. '.jpg', '.mp3')
+
+    Returns:
+        A sanitized filename string in the format: key_step_id.extension
+    """
+    # Avoid writing to absolute paths by striping any leading slashes.
+    # The key has already been validated for windows operating systems in util.check_windows_valid_filename
+    # This ensures the key does not contain invalid characters for windows, such as '\' or ':'.
+    # So we can check only for '/' in the key.
+    key = str(key).lstrip("/")
+
+    # Avoid directory traversal by replacing dots with underscores.
+    keys = key.split("/")
```

Comment:
you only split on / because \ is not allowed on windows?

oh the media code is so convoluted :( 
so does it mean that on windows we don't allow to specify nested paths?

Reply:
Thats correct, since \ is not allowed a user cannot provide an absolute or even relative path.

With this change now, since I split by / then recombine the key with 'os.sep' it will create the media files under a nested folder structure when defining a key like '/some/media/file'
I verified this behavior on my own windows machine

Reply:
my  only thinking is that there is some implicit assumption about a logic that comes from a different place, it seems to me that we should treat it the same in both windows and no windows cases

Reply:
Good point, I have instead made the change to convert any `/` to `os.sep` for windows. Then the rest of the logic is the same across platforms.